GenericTensor/Core/Tensor/Iterate.cs

```diff
@@ -48,11 +48,12 @@ private void NextIndex(int[] indecies, int id)
         /// <summary>
         /// Iterate over array of indecies and a value in TPrimitive
         /// </summary>
-        public IEnumerable<(int[] index, T value)> Iterate()
+        public IEnumerator<(int[] Index, T Value)> GetEnumerator()
         {
             foreach (var ind in IterateOver(0))
                 yield return (ind, this.GetValueNoCheck(ind));
         }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => GetEnumerator();
```

Comment:
There should be no implicit iterators, everything should be as explicit as possible for this lib

Reply:
Non-idiomatic but OK